libs/file-storage/src/lib/controller/abstract-file-storage.controller.ts

```diff
@@ -0,0 +1,57 @@
+import { Body, HttpCode, HttpStatus, Post, Res } from '@nestjs/common';
+import { AbstractFileService } from '../services';
+import { FileDownloadRequest } from './vo/file-donwload.dto';
+import { PreAssignRequest, PreAssignResponse } from './vo/pre-assign.dto';
+import { FastifyReply } from 'fastify';
+import 'reflect-metadata';
+
+export abstract class AbstractFileStorageController {
+  protected constructor(
+    private readonly bucket: string,
+    private readonly fileService: AbstractFileService,
+  ) {}
+
+  @Post('download-file')
+  @HttpCode(HttpStatus.MOVED_PERMANENTLY)
+  protected async downloadFlyerFilesFromAWS(
+    @Res() reply: FastifyReply,
+    @Body() fileDownloadRequest: FileDownloadRequest,
+  ) {
+    const { key } = fileDownloadRequest;
+
+    const url = await this.fileService.generateDownloadFilePreSignUrl(
+      this.bucket,
+      key,
+      300,
+    );
+
+    reply
+      .status(301)
+      .header('Content-Security-Policy', "img-src 'self'")
+      .redirect(url);
+  }
+
+  @Post('get-pre-assign-url')
+  @HttpCode(HttpStatus.OK)
+  protected async getPreAssignUrl(
+    @Body() preAssignRequest: PreAssignRequest,
```

Comment:
is it for download or upload? I think it make sense to be more obvious in naming